colossalai/zero/gemini/gemini_ddp.py

```diff
@@ -318,6 +331,7 @@ def backward(self, loss: torch.Tensor):
         self._post_backward()
 
     def backward_by_grad(self, tensor, grad):
+        self._pre_backward()
```

Comment:
`backward_by_grad()` shoud not be called, since Gemini is not compatible with pipeline. Raise a runtime error here.

Reply:
updated.